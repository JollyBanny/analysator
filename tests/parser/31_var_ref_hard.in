type Point = record
    x: integer;
    y: integer;
end;

type PointArr = array [0..2] of Point;

var cursor: Point;

function AwesomeFunc(cursor_: Point): PointArr;
var arr: PointArr;
begin
    writeln('asdas');
end;

begin
    AwesomeFunc(cursor);
end.