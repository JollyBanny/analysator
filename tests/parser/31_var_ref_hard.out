program
└──── program block
      ├──── TYPE
      │     └──── type declaration
      │           ├──── point
      │           └──── RECORD
      │                 ├──── :
      │                 │     ├──── x
      │                 │     └──── integer
      │                 └──── :
      │                       ├──── y
      │                       └──── integer
      ├──── TYPE
      │     └──── type declaration
      │           ├──── pointarr
      │           └──── ARRAY
      │                 ├──── ..
      │                 │     ├──── 0
      │                 │     └──── 2
      │                 └──── point
      ├──── VAR
      │     └──── var declaration
      │           ├──── cursor
      │           └──── point
      ├──── FUNCTION
      │     ├──── header
      │     │     ├──── awesomefunc
      │     │     ├──── parameter
      │     │     │     ├──── cursor_
      │     │     │     └──── point
      │     │     └──── pointarr
      │     └──── block
      │           ├──── VAR
      │           │     └──── var declaration
      │           │           ├──── arr
      │           │           └──── pointarr
      │           └──── Compound statement
      │                 └──── Call statement
      │                       └──── writeln
      │                             └──── 'asdas'
      └──── Compound statement
            └──── Call statement
                  └──── awesomefunc
                        └──── cursor
